app/models/concerns/pathways_version_three_calculations.rb

```diff
@@ -43,6 +43,48 @@ def assessment_type_options
       }
     end
 
+    def locked?
+      return false if lockable_fields.empty?
+      return false if locked_until.blank?
+      return false if Date.current >= locked_until
+
+      ! in_lock_grace_period?
+    end
+
+    def lockable_fields
+      return [] unless assessment_type == pathways_assessment_type.to_s
+
+      [
+        :days_homeless_in_the_last_three_years,
+        :additional_homeless_nights,
+        :total_days_homeless_in_the_last_three_years,
+      ]
+    end
+
+    def lock
+      return if locked?
+      return if lockable_fields.empty?
+
+      lock_days = Config.get(:lock_days)
+      return if lock_days.zero?
+      return if in_lock_grace_period?
+
+      self.locked_until = Date.current + lock_days.days
+    end
+
+    def in_lock_grace_period?
+      return true if lockable_fields.empty?
+      return false if locked_until.blank?
```

Comment:
I don't know if these should be true or false, but it seems like they should be the same? Maybe false? You can't be in the grace period if there isn't one, or there are no fields?

Reply:
If `locked_until` is blank, we can't be in the grace period or it prevents setting the lock date in `lock`